internal/core/server/project/ProjectManager.ts

```diff
@@ -816,6 +816,17 @@ export default class ProjectManager {
 		return this.projectDirectoryToProject.get(path);
 	}
 
+	/**
+	 * Given a path, it returns the list of projects
+	 * @param path
+	 */
+	public getProjectHierarchyFromPath(
+		path: AbsoluteFilePath,
+	): Array<ProjectDefinition> {
+		const found = this.assertProjectExisting(path);
+		return this.getHierarchyFromProject(found);
+	}
```

Comment:
Nice! I think the problem is that we are in the process of adding a `package.json` that contains a project config, and during validation of the `package.json` we haven't loaded the project yet.

We can fix it with:

```suggestion
	public getProjectHierarchyFromPath(
		path: AbsoluteFilePath,
	): Array<ProjectDefinition> {
		const found = this.findLoadedProject(path);
		if (found === undefined) {
			return [];
		} else {
  			return this.getHierarchyFromProject(found);
  		}
	}
```

This means that `package.json` files with a project config that contains a rule referencing themselves wont work. We can make the project config initialization happen before we normalize the manifest, but it would be a lot of work to do so I think this is fine as a temporary fix.